src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -919,6 +920,29 @@ void ov::npuw::JustInferRequest::unsafe_run_this_prep_next(std::size_t idx, bool
     }  // if (replaced_by)
 }
 
+ov::Tensor ov::npuw::JustInferRequest::allocTensor(const ov::element::Type type,
+                                                   const ov::Shape& shape,
+                                                   const std::string& device) {
+    if (!m_alloc_required || device == "CPU") {
+        return ov::Tensor(type, shape);
+    }
```

Comment:
this _and_ having `device="NPU"` by default is looks a very obscure way to allocate the tensors in the right region.

So you're using this method in tree contexts:
1. For function results - where `device` is taken into account
2. For global inputs, where `device` is not passed and defaults to `"NPU"` (meh)
3. For global results, same?

Also the logic completely discards the `WEIGHTS_BANK_ALLOC` setting we have here. OK, this is not weights, but that flag exists for a reason you know. Why do we think we bypass *that* problem here?

A clearer way how it could've been done:
1. No default arguments here - just always pass the device for yourself
2. Add two methods to the `CompiledModel` so frame the decision making logic:
   * `std::string global_mem_device()`
   * `std::string funcall_mem_device(idx)`

The first only takes device distribution into account.
The second takes the subgraph affinity into account (you can access it).
BOTH should also take the BANK_ALLOC into account if that's set.

Reply:
Done